src/java/org/apache/cassandra/db/CassandraKeyspaceWriteHandler.java

```diff
@@ -61,6 +64,43 @@ public WriteContext beginWrite(Mutation mutation, boolean makeDurable) throws Re
         }
     }
 
+    private CommitLogPosition addToCommitLog(Mutation mutation)
+    {
+        CommitLogPosition position;
+        // Usually one of these will be true, so first check if that's the case.
+        boolean allSkipCommitlog = true;
+        boolean noneSkipCommitlog = true;
+        for (TableId id : mutation.getTableIds())
+        {
+            if (keyspace.getColumnFamilyStore(id).writesShouldSkipCommitLog())
+                noneSkipCommitlog = false;
+            else
+                allSkipCommitlog = false;
+        }
+
+        if (!noneSkipCommitlog)
+        {
+            if (allSkipCommitlog)
+                return null;
+            else
+            {
+                Set<TableId> ids = new HashSet<>();
+                for (TableId id : mutation.getTableIds())
+                {
+                    if (keyspace.getColumnFamilyStore(id).writesShouldSkipCommitLog())
+                        ids.add(id);
+                }
+                mutation = mutation.without(ids);
```

Comment:
I strongly encourage using names which carry semantics information.


E.g. let's consider `without`. 
It looks important that the mutation is not modified in place. 
- Is `Mutation` class immutable? It's not clear. 
- It isn't clear whether `without` creates a new mutation, or acts as a builder returning `this`. One has to peek into the implementation. 


Naming the method in a way that cleary specifies the intent allows making assumptions easier. Like - reading `cloneWithout` or `newWithout` or some similar name makes it obvious that the code author did not forget about the issue. Also, if the implementation does not agree with the name it is obvious we've got a bug.


It would be even better if we could express immutability of `mutation` instance somehow, but java :shrug:


Alone, it's a tiny, insignificant issue. Please bear with me, but when I'm reading some code trying to find a bug it gets frustrating after a dozen of such :/

Reply:
Point taken, but this is outside the scope of the ticket. STAR-205 may give an opportunity to correct this.